wireguard-vpn/js/MainController.js

```diff
@@ -304,18 +304,26 @@ Ext.define('Ung.apps.wireguard-vpn.MainController', {
 
     getNextUnusedPoolAddr: function() {
         var me = this,
-            vm = me.getViewModel(),
-            addressPool = vm.get('settings.addressPool'),
-            store = vm.getStore('tunnels'),
-            pool = addressPool.split("/")[0];
-
+        vm = me.getViewModel(),
+        addressPool = vm.get('settings.addressPool'),
+        store = vm.getStore('tunnels');
+        var subnetRange = Util.calculateSubnetRange(addressPool);
+        var pool = subnetRange.pool;
+        var lastAddress = subnetRange.lastAddress;
         // Assume the first pool address is used by the wg interface
         var nextPoolAddr = Util.incrementIpAddr(pool, 2);
-        while (Util.isAddrUsed(nextPoolAddr, store, 'peerAddress')) {
+        // Convert IP addresses to integers for comparison
+        var nextPoolAddrInt = Util.ipToInt(nextPoolAddr);
+        var lastAddressInt = Util.ipToInt(lastAddress);
+        while (nextPoolAddrInt <= lastAddressInt && (Util.isAddrUsed(nextPoolAddr, store, 'peerAddress'))) {
             nextPoolAddr = Util.incrementIpAddr(nextPoolAddr, 1);
+            nextPoolAddrInt = Util.ipToInt(nextPoolAddr);
+        }
+        if (nextPoolAddrInt <= lastAddressInt) {
+            return nextPoolAddr;
+        } else {
+            return '';
```

Comment:
In general practice we used to keep the value as blank instead of null. https://github.com/untangle/ngfw_src/blob/486311d899014f723d168c7389f720e40e3b9b76/wireguard-vpn/js/view/Tunnels.js#L48